src/core/linter-rules/check-punctuation.js

```diff
@@ -31,6 +31,9 @@ function lintingFunction(conf, doc) {
   const offendingElements = Array.from(doc.querySelectorAll("p:not(#back-to-top)"))
     .filter((elem) => !(punctuatingRegExp.test(elem.textContent)));
   
+  //To help the Editor reach he offending element at once
+  console.log(offendingElements);
```

Comment:
So, the linter already does this for us here:
https://github.com/w3c/respec/blob/develop/src/core/linter.js#L60

The actual bug is to figure out why the Linter is not doing this right now. I suspect there is something wrong with what this linter rule is returning.

Reply:
I guess its working. Is this not what you are talking about?

![screenshot from 2018-04-06 09-27-00](https://user-images.githubusercontent.com/25258877/38402817-b2850816-397c-11e8-839d-d78193194173.png)

Reply:
Ah, ok, thanks for confirming. Wonder why that didn't come up when I was trying it out.

Reply:
I guess what is then left is to figure out why the error dialog is not rendering the markdown properly.

Reply:
> I guess what is then left is to figure out why the error/warning dialog is not rendering the markdown properly.

yep! I'll look into that.

> . Wonder why that didn't come up when I was trying it out.

Maybe because no punctuation error existed. The ones present here are the ones I have put from the test files.

Reply:
Ah, if it's showing an error when there are no invalid `p`s, then that's a bug. We should have added a test for that.

Reply:
Oh, those `</p>` are what are screwing everything up. My editor must have added them by accident.

Reply:
oh wait, I got confused. We want everything to show up like the other linter warnings (on one line).

Reply:
> As we already have an open <p> tag in this statement, the next time we add a <p> tag it first closes the pre-existing <p> tag. And thus, it justifies </p><p>. I guess his might be one reason.

We shouldn't have any stray `<p>`, so the bug is that we didn't escape them :)

Reply:
How do we escape ? 
```
en: {
    description: "`\<p\>` tags should end with a punctuation mark.",
    howToFix: `Please make sure \`\<p\>\` tags end with one of: ${humanMarks}.`,
  }
```

This isn't working :(